api/dbadapters/bigquery.ts

```diff
@@ -96,22 +98,57 @@ export class BigQueryDbAdapter implements IDbAdapter {
       .promise();
   }
 
-  public async evaluate(statement: string) {
-    try {
-      await this.getClient().query({
-        useLegacySql: false,
-        query: statement,
-        dryRun: true
-      });
-      return dataform.QueryEvaluationResponse.create({
-        status: dataform.QueryEvaluationResponse.QueryEvaluationStatus.SUCCESS
-      });
-    } catch (e) {
-      return dataform.QueryEvaluationResponse.create({
-        status: dataform.QueryEvaluationResponse.QueryEvaluationStatus.FAILURE,
-        error: parseBigqueryEvalError(e)
-      });
+  public async evaluate(
+    queryOrAction: string | dataform.Table | dataform.Operation | dataform.Assertion,
+    projectConfig: dataform.IProjectConfig
+  ) {
+    let executionTasks: dataform.ExecutionTask[];
+    if (typeof queryOrAction === "string") {
+      executionTasks = [dataform.ExecutionTask.create({ statement: queryOrAction })];
+    } else {
+      try {
+        const coreAdapter = new BigQueryAdapter(projectConfig, version);
+        if (queryOrAction instanceof dataform.Table) {
+          executionTasks = coreAdapter
+            .publishTasks(
+              queryOrAction,
+              { useSingleQueryPerAction: projectConfig?.useSingleQueryPerAction },
+              null
+            )
+            .build();
+        } else if (queryOrAction instanceof dataform.Operation) {
+          executionTasks = queryOrAction.queries.map(statement =>
```

Comment:
i think the last comment maybe was confusing? i mean that we should change (back to what you had originally) evaluate to return an array, so that cases like this make sense.

Reply:
Sure, have updated